ext/dba/dba.c

```diff
@@ -525,42 +561,45 @@ static void php_dba_open(INTERNAL_FUNCTION_PARAMETERS, bool persistent)
 		RETURN_THROWS();
 	}
 
+	char *resource_key;
+	size_t resource_key_len = spprintf(&resource_key, 0,
+		"dba_%d_%s_%s_%s", persistent, ZSTR_VAL(path), ZSTR_VAL(mode), handler_str ? ZSTR_VAL(handler_str) : ""
+	);
```

Comment:
Why not `strpprintf()`/`zend_strpprintf()` to get a `zend_string` directly?

Reply:
I copy-pasted this piece of code from ext/odbc :) But I didn't really know about these APIs to be honest, so that's the main reason.

Reply:
BTW we sometimes need to create persistent strings, which this API doesn't support as far as I saw, so at the end of the day, we would have to create a new `zend_string` when dealing with persistent resources.